### PR TITLE
Bring consistency to download and upload APIs

### PR DESCRIPTION
**Pull request by muellerzr**
Closes https://github.com/huggingface/huggingface_hub/issues/47

This PR aims to bring some consistency in how the api can be used, by letting a user both pass in a specified `organization` as well as using `organization/model_name` for functions revolving around uploading and downloading

### CODE REVIEW

**Comment by muellerzr** [comment]
@osanseviero mostly interested on if you think I missed any. There weren't many instances of this happening from a user perspective I could *see*, but one may have trickled past me

File: src/huggingface_hub/hf_api.py [hunk 1]
```diff
@@ -811,6 +811,15 @@ def create_repo(
...
             raise ValueError("Invalid repo type")
 
+        if "/" in name:
+            if organization is not None:
+                if organization != name.split("/")[1]:
+                    raise ValueError(
+                        f"Both a `name` with an organization and an `organization` were passed in that do not align ({name}, {organization}). Please ensure these align to the same organization"
+                    )
+                else:
+                    name, organization = name.split("/")
```
Comment (osanseviero) [inline src/huggingface_hub/hf_api.py:821]: I would actually do the split first and then check if length is 2 (avoid osan/repo/anotherthing) and raise a more informative value error

**Review by osanseviero — COMMENTED** [review]
Overall looks good. I think you cover all the cases

**Review by LysandreJik — APPROVED** [review]
Agree with @osanseviero's comment, otherwise this looks good!

**Review by osanseviero — APPROVED** [review]

**Comment by muellerzr** [comment]
Based on advice from @LysandreJik I forgo that logic for a complete refactor, making use of the `repo_type_and_id_from_hf_id` functionality, since we already have a function that *can*  check if it's a valid namespace.

File: src/huggingface_hub/hf_api.py [hunk 2]
```diff
@@ -808,6 +808,17 @@ def create_repo(
...
+        checked_name = repo_type_and_id_from_hf_id(name)
+        name = checked_name[2]
+
+        if organization is None:
+            organization = checked_name[1]
+        elif organization != checked_name[1]:
+            raise ValueError(
+                f"""Passed `organization` and `name` organization are not the same ({organization}, {checked_name[1]}).
+            Please either include the organization in only `name` or the `organization` parameter, such as `api.create_repo({checked_name[0]}, organization={organization})` or `api.create_repo({checked_name[1]}/{checked_name[2]})`"""
+            )
```
Comment (LysandreJik) [inline src/huggingface_hub/hf_api.py:820]: Nice

**Review by LysandreJik — APPROVED** [review]
Looks good to me!

**Pull request merged**